src/lib/core/plugin.js

```diff
@@ -160,8 +160,12 @@ Plugin.prototype.addContractFile = function(file) {
   this.addPluginType('contractFiles');
 };
 
-Plugin.prototype.registerConsoleCommand = function(cb) {
-  this.console.push(cb);
+Plugin.prototype.registerConsoleCommand = function(optionsOrCb) {
+  if (typeof optionsOrCb === 'function') {
+    this.logger.warn(__('Registering console commands with a function is deprecated'));
```

Comment:
Just to be clear: it'll be considered deprecated, but it still works. So maybe we can make the clear in the deprecation message and say something like:

> Registering console commands with function syntax is deprecated and will likely be removed in future versions of Embark.